README.md

```diff
@@ -1,42 +1,89 @@
-# kilt-extension-api
+# KILT Extension API
 
-KILT Extension helper functions. The tools you need to add a KILT extension to your app, or add KILT functionality to your extension.
+KILT Extension API is a JavaScript/TypeScript library that provides helper functions for interacting with KILT enabled extensions.
+It facilitates seamless communication between your application and KILT extensions.
 
-## Get Extension
+## Getting Started
 
-In order to use the extension with KILT credential API the extension needs to be injected into the application close to the loading of the application. Here you can use `getExtensions` function to add it directly to the application and wait for its response.
+Before you can communicate with KILT extensions, you must call the `initializeKiltExtensionAPI()` function to signal the API versions supported by your application.
+This is crucial for the extension to inject the appropriate scripts into the website.
 
-Now you can access the extensions API and handle the data and interactions between the Dapp and Extension.
+```js
+import { initializeKiltExtensionAPI } from 'kilt-extension-api'
 
-See the `react-example.ts` example inside the get extension folder, here is how you can use `getExtensions` in a react app?
+initializeKiltExtensionAPI()
+```
+
+## Get Extensions
+
+The `getExtensions()` function returns a list of extensions currently injected into the website.
+
+```js
+import { getExtensions } from 'kilt-extension-api'
 
-## Inject Extension
+const extensions = getExtensions()
+```
+
+## Watch Extensions
+
+Extensions may take longer to load than the website.
+Therefore, the first call to `getExtensions()` might not return all available extensions.
+To receive updates on additional extensions as they load, you can use `watchExtensions`.
+
+Here's an example of how you can use this function in a React application:
+
+```js
+import { watchExtensions, Types } from 'kilt-extension-api'
+
+export default function Home(): JSX.Element {
+  const [extensions, setExtensions] = useState<
+    Types.InjectedWindowProvider<Types.PubSubSessionV1 | Types.PubSubSessionV2>[]
+  >([])
+
+  useEffect(() => {
+    watchExtensions((extensions) => {
+      setExtensions(extensions)
+    })
+  }, [])
+
+  return (
+    <>
+      <h2>Extensions</h2>
+      <ul>
+        {extensions.map((ext, i) => (
+          <li key={i}>{ext.name}</li>
+        ))}
+      </ul>
+    </>
+  )
+}
```

Comment:
Do you want to add TS instead, or maybe remove all the types?
```suggestion
```ts
import { watchExtensions, Types } from 'kilt-extension-api'

export default function Home(): JSX.Element {
  const [extensions, setExtensions] = useState<
    Types.InjectedWindowProvider<Types.PubSubSessionV1 | Types.PubSubSessionV2>[]
  >([])

  useEffect(() => {
    watchExtensions((extensions) => {
      setExtensions(extensions)
    })
  }, [])

  return (
    <>
      <h2>Extensions</h2>
      <ul>
        {extensions.map((ext, i) => (
          <li key={i}>{ext.name}</li>
        ))}
      </ul>
    </>
  )
}
```

Reply:
[ed51ec2](https://github.com/KILTprotocol/kilt-extension-api/pull/15/commits/ed51ec28238b81c2ee247d629ca98ea1f6fb9e2d)